qmetaobject/src/qtdeclarative.rs

```diff
@@ -201,6 +202,106 @@ impl Default for QQuickView {
     }
 }
 
+#[repr(u32)]
+#[derive(Clone, Copy)]
+pub enum CompilationMode {
+    PreferSynchronous,
+    Asynchronous,
+}
+
+#[repr(u32)]
+#[derive(Clone, Copy)]
+pub enum ComponentStatus {
+    Null,
+    Ready,
+    Loading,
+    Error
+}
+
+cpp!{{
+    struct QQmlComponentHolder {
+        std::unique_ptr<QQmlComponent> component;
+
+        QQmlComponentHolder(QQmlEngine *e) : component(new QQmlComponent(e)) {}
+    };
+}}
+
+cpp_class!(
+    /// Wrapper for QQmlComponent
+    pub unsafe struct QmlComponent as "QQmlComponentHolder"
+);
+
+impl QmlComponent {
+    /// Create a QmlComponent using the QmlEngine.
+    pub fn new(engine: &QmlEngine) -> QmlComponent {
+        unsafe {
+            cpp!([engine as "QmlEngineHolder*"] -> QmlComponent as "QQmlComponentHolder" {
+                return QQmlComponentHolder(engine->engine.get());
+            })
+        }
+    }
+
+    pub fn get_cpp_object(&self) -> *mut c_void {
```

Comment:
Could also have some documentation.
(Also explain what's the relation with the method in the QObject trait)